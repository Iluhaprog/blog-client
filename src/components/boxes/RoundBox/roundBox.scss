@mixin size($w, $h) {
    max-width: $w;
    width: 100%;
    height: $h;
}

.round-box {
    border-radius: 50%;
    overflow: hidden;
    @include size(65px, 65px);

    &__max {
        @include size(300px, 300px);
    }

    * {
        width: 100%;
        height: 100%;
    }

    img {
        object-position: center;
        object-fit: cover;
    }
}