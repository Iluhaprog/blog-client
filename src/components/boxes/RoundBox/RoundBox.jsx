import './roundBox.scss';

export default ({ children, max }) => (
    <div className={'round-box' + (max ? ' round-box__max' : '')}>
        { children }
    </div>
);