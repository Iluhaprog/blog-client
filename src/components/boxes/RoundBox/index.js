import RoundBox from './RoundBox';

export { RoundBox };