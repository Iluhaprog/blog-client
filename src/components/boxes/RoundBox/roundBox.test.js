import { render } from '@testing-library/react';
import RoundBox from './RoundBox';

test('Test view of round image component', () => {
    const  { asFragment } = render(<RoundBox></RoundBox>);
    expect(asFragment()).toMatchSnapshot();
});

test('Test view of round image max size component', () => {
    const  { asFragment } = render(<RoundBox max></RoundBox>);
    expect(asFragment()).toMatchSnapshot();
});