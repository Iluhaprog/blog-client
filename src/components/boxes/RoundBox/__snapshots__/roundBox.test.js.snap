// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test view of round image component 1`] = `
<DocumentFragment>
  <div
    class="round-box"
  />
</DocumentFragment>
`;

exports[`Test view of round image max size component 1`] = `
<DocumentFragment>
  <div
    class="round-box round-box__max"
  />
</DocumentFragment>
`;
